0000-settlement-engines/0000-settlement-engines.md

```diff
@@ -0,0 +1,491 @@
+
+In order to continue transacting, the peer must settle their liabilities. In most cases, this is accomplished through sending a payment on a settlement system that both peers have agreed to use. The connector should acknowledge the incoming payment&mdash;irrevocably discharging some liability the peer owed to it&mdash;and enabling it to clear and forward subsequent packets from the peer on credit.
+
+Settlement engines provide a standardized mechanism for Interledger connectors to coordinate and reconcile these settlements.
+
+### Accounting
+
+Connectors MUST record their transactions with peers and/or the effects of their transactions with peers through the process of accounting.
+
+In this accounting context, an **account** represents amounts received (credits) and amounts owed (debits) for a set of transactions between counterparties. The **balance** of an account is the net difference between these credits and debits. All the balances and transactions of an account are denominated in a single, fungible asset.
+
+Interledger connectors are RECOMMENDED to operate an **accounting system** which keeps a record of two accounts for each peer:
+
+- **Accounts payable**, the amount owed by the connector to its peer for packets its peer has fulfilled.
+  - Positive amount indicates the connector is indebted to its peer (a _liability_ for the connector).
+  - Negative amount indicates the connector has sent a pre-payment to its peer.
+- **Accounts receivable**, the amount owed to the connector by its peer for packets the connector has fulfilled.
+  - Positive amount indicates its peer is indebted to the connector (an _asset_ to the connector).
+  - Negative amount indicates its peer has sent a pre-payment to the connector.
+
+(Note that "accounts payable" and "accounts receivable" refer to accounts as _records_, distinct from accounts referring to the _arrangement_ between counterparties.)
+
+Thus, the connector's accounts payable with its peer should mirror its peer's accounts receivable with the connector, and respectively, the connector's accounts receivable should equal its peer's accounts payable.
+
+### Settlement
+
+A **settlement** is the irrevocable discharge of a liability by providing something of value to the party to whom the liability is owed.
+
+Settlements may occur on a **settlement system**, or medium for exchanging value. Some settlements may transfer funds on a **ledger**, or registry of account balances and/or transactions, which is a type of settlement system. (Although not all settlement systems are ledgers, here, the terms are sometimes used interchangeably.)
+
+Examples of settlement systems include:
+
+- Bank clearing houses
+- Credit card processors
```

Comment:
Lol, yeah, "banking clearing houses" and "credit card processors" may be confusing since they're clearing systems, yet they're still kinda-sorta a settlement system with respect to Interledger. Should we include them anyways? Are there any other useful examples that are missing? @adrianhopebailie @sappenin